test/spec/functional/viewTransition.js

```diff
@@ -0,0 +1,59 @@
+var testSrcPath = '../../source',
+    spyOnBackbone = require('./backboneSpy');
+
+describe('A View that can transition in tracked views', function() {
+
+  var env, _, $, ParentView;
+
+  beforeEach(function(done) {
+    require('./clientEnv')().done(function(environment) {
+      env = environment;
+      $ = env.window.$;
+      _ = env.window._;
+      ParentView = require(testSrcPath + '/transitionUsingParentViewGenerator')(env.window);
+      $('body').append('<div class="app"></div>');
+      done();
+    });
+  });
+
+  it('can transition in the first view into an injection site', function(done) {
+    var parentView = new ParentView();
+    parentView.render();
+    expect(parentView.$el).toBeDefined();
+    parentView.transitionPromise.done(function() {
```

Comment:
great question - one that I considered. I didn't want to complicate the process.

But, I gave it a go. prerender, postrender, and attachTrackedViews all can return a promise (or promises) if they want or not and render will return one promise that is resolved when all of them are resolved.